plugins/dig-now.cpp

```diff
@@ -835,3 +841,14 @@ DFhackCExport command_result plugin_init(color_ostream &,
 DFhackCExport command_result plugin_shutdown(color_ostream &) {
     return CR_OK;
 }
+
+// External API
+
+// runs dig-now for the specified tile. default options apply.
+bool dig_now_tile (color_ostream& out, const DFCoord& pos)
```

Comment:
This does seem ideal actually as the user can just iterate over themselves.

Reply:
Copying from discord: the helper function you can now define to call the api is this:

// executes dig designations for the specified tile coordinates
static bool dig_now_tile(color_ostream &out, const DFCoord& pos) {
    auto L = Lua::Core::State;
    Lua::StackUnwinder top(L);

    if (!lua_checkstack(L, 2) ||
            !Lua::PushModulePublic(out, L, "plugins.dig-now", "dig_now_tile"))
        return false;

    Lua::Push(L, pos);

    if (!Lua::SafeCall(out, L, 1, 1))
        return false;

    return lua_toboolean(L, -1);
}

Reply:
I guess I'll be merging this soon.
So this will work upon merge?

Reply:
It should, yeah. Tell me if you have any trouble with it